### PR TITLE
Update TensorRT-LLM main branch

### PR DESCRIPTION
**Pull request by kaiyux**
* Features
  - Baichuan 
    - P-tuning support
    - INT4-AWQ and INT4-GPTQ support
  - The batch manager can now return logits from context and generation phases
  - Support for combining `repetition_penalty` and `presence_penalty` #274
  - Support for `frequency_penalty` #275
* Bug fixes
  - Fix LLaMA with LoRA enabled build failure #673
  - Fix InternLM SmoothQuant build failure #705
  - Fix Bloom int8_kv_cache functionality  #741
* Documentation
  - Batch manager arguments documentation updates
  - Add v0.7 change log

### CODE REVIEW

**Review by Shixiaowei02 — APPROVED** [review]

**Comment by shatealaboxiaowang** [comment]
> * Features
>   
>   * Baichuan
>   * P-tuning support
>   * INT4-AWQ and INT4-GPTQ support
>   * The batch manager can now return logits from context and generation phases
>   * Support for combining `repetition_penalty` and `presence_penalty` [Support for combining `repetition_penalty`, `presence_penalty` #274](https://github.com/NVIDIA/TensorRT-LLM/issues/274)
>   * Support for `frequency_penalty` [Support for `frequency_penalty` #275](https://github.com/NVIDIA/TensorRT-LLM/issues/275)
> * Bug fixes
>   
>   * Fix LLaMA with LoRA enabled build failure [Failed to build Llama-2-13b-hf engine with Lora adapter #673](https://github.com/NVIDIA/TensorRT-LLM/issues/673)
>   * Fix InternLM SmoothQuant build failure [[0.6.1] InternLM SmoothQuant does not work #705](https://github.com/NVIDIA/TensorRT-LLM/issues/705)
>   * Fix Bloom int8_kv_cache functionality  [【bloom】convert_checkpoint.py  local variable 'int8_weights' referenced before assignment #741](https://github.com/NVIDIA/TensorRT-LLM/issues/741)
> * Documentation
>   
>   * Batch manager arguments documentation updates
>   * Add v0.7 change log

 Can directly support INT4-AWQ and INT4-GPTQ ? such as [CodeLlama-13B-AWQ(https://huggingface.co/TheBloke/CodeLlama-13B-AWQ) ?

**Comment by kaiyux** [comment]
> Can directly support INT4-AWQ and INT4-GPTQ ? such as [CodeLlama-13B-AWQ(https://huggingface.co/TheBloke/CodeLlama-13B-AWQ) ?

@shatealaboxiaowang The indentation was incorrect, and the update should be P-tuning and INT4-AWQ and INT4-GPTQ support for Baichuan model. Thanks very much for following the work!
